controllers/topics/create.js

```diff
@@ -0,0 +1,37 @@
+const {Topics} = require('../../databases/models');
+const {insertTopics} = require('../../utils/post');
+
+module.exports = async (req, res) => {
+  try {
+    let {name} = req.body;
+    name = name.toLowerCase();
+    let topics = await Topics.findOne({
+      where: {
+        name,
+        deleted_at: null
+      }
+    });
+
+    if (topics) {
+      return res.status(400).json({
+        code: 400,
+        status: 'failed',
+        message: 'Topic already exists'
+      });
+    }
+
+    insertTopics([name]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There's a potential issue here. The `insertTopics` function is called but its result or possible errors are not handled. If there's an error during insertion, it won't be caught and the function will still return a success message. Also, `insertTopics` is an asynchronous function and should be awaited.

```diff
-    insertTopics([name]);
+    await insertTopics([name]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->